public/app/plugins/datasource/graphite/utils.ts

```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash';
+
+/**
+ * Graphite-web before v1.6 returns HTTP 500 with full stack traces in an HTML page
```

Comment:
@ifrost I wonder isn't there something like a support for accept header? Like if send the request with accept header application/json wouldn't it return better error? I think lots of times I seen this because the API defaulted to html response is it assumed somebody just pasted the request to browser.

Reply:
I'll double check but as far as I'm aware by default graphite-web returns HTTP 500 with the HTML for any unhandled Django error regardless to headers. It can be customised as for example [here](https://stackoverflow.com/questions/9261830/django-conditional-500-response-depending-on-accept-header), but in graphite-web it's configured to simply return[ an HTML page](https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/views.py). What has changed in 1.6 is that 4XX is returned instead of 500  ([change](https://github.com/graphite-project/graphite-web/pull/2467)) with a lot of extra error handling and much [simpler text-based error message](https://github.com/graphite-project/graphite-web/pull/2467/files#diff-95b0aaddd23366edf1c321630a1d1e84913ecb5a9bcceb9927a33767d35e507cR50) that doesn't need any extra parsing.